src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/PostgreSqlEngine.java

```diff
@@ -832,4 +776,46 @@ protected ResultIterator createResultIterator(PreparedStatement ps) throws Datab
     protected QueryExceptionHandler getQueryExceptionHandler() {
         return PG_QUERY_EXCEPTION_HANDLER;
     }
+
+    /**
+     * Helper method that logs the time while the Postgres {@link Statement update query} is being executed.
+     * <br>
+     * This will help identify cases via logs where Postgres database is already locked due to an external query being
+     * executed and the current one is waiting for that one to finish first.
+     *
+     * @param query The query in {@link String} format which will be executed.
+     * @throws SQLException If the {@link Statement query} fails during its execution.
+     * @implNote It prints a log every 5 minutes so the user is aware that the query is still being executed.
+     */
+    private int executeUpdateQuery(final String query) throws SQLException {
+        final long startMilliSeconds = System.currentTimeMillis();
+        final Runnable printLog = () -> {
+            final Duration timePassed = Duration.of(System.currentTimeMillis() - startMilliSeconds, ChronoUnit.MILLIS);
```

Comment:
I would change this to 
`long timePassed = Duration.ofMillis(System.currentTimeMillis()).minusMillis(startMilliSeconds).toMinutes();`

and then use it directly in the logs below.

Reply:
Done!